### PR TITLE
List all sets pt II: DRY

### PR DESCRIPTION
**Pull request by ialarmedalien**
Dry out the test code a little by moving all the setup code into a separate file.

### CODE REVIEW

File: .github/workflows/kb_sdk_test.yaml [hunk 1]
```diff
@@ -4,11 +4,14 @@ on:
     push:
         branches:
             - master
+            - main
```
Comment (ialarmedalien) [inline .github/workflows/kb_sdk_test.yaml:7]: Will rename the deployed branch to 'main' as part of this work.

File: .github/workflows/kb_sdk_test.yaml [hunk 2]
```diff
@@ -4,11 +4,14 @@ on:
...
 
     pull_request:
-        branches:
-            - master
-            - develop
+        types:
```
Comment (ialarmedalien) [inline .github/workflows/kb_sdk_test.yaml:10]: run on all PRs of these types (copied from SN repo)

File: .github/workflows/kb_sdk_test.yaml [hunk 3]
```diff
@@ -54,8 +57,13 @@ jobs:
...
               shell: bash
-              env:
-                  CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
               run: |
                   sh $GITHUB_WORKSPACE/kb_sdk_actions/bin/kb-sdk test
-                  bash <(curl -s https://codecov.io/bash)
+
+            - name: Send to Codecov
+              id: send_to_codecov
+              uses: codecov/codecov-action@v3
```
Comment (ialarmedalien) [inline .github/workflows/kb_sdk_test.yaml:65]: updated way to invoke the codecov spell

File: scripts/run_async.sh
Comment (briehl) [inline scripts/run_async.sh:1]: Is this a new file? I coulda sworn there was one there already. Oh, are permissions just changed, and it gets filled out?

File: test/test_config.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    cfg_dict = {}
+    config = ConfigParser()
+    config.read(config_file)
+    for nameval in config.items("SetAPI"):
+        cfg_dict[nameval[0]] = nameval[1]
+    return cfg_dict
+
+
+def get_test_config():
+    conf = parse_config()
```
Comment (briehl) [inline test/test_config.py:24]: A docstring with the expected returns would be helpful here.

File: test/test_config.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+    # set up the SetAPI implementation
+    set_api_client = SetAPI(conf)
+
+    # create a workspace for the tests
+    suffix = int(time.time() * 1000)
+    ws_name = "test_SetAPI_" + str(suffix)
+    ws_url = conf['workspace-url']
+    ws_client = Workspace(ws_url, token=token)
+    ws_client.create_workspace({'workspace': ws_name})
```
Comment (briehl) [inline test/test_config.py:54]: The config creates its own workspace here, maybe adding a `config.tear_down()` function or something to handle deletion would be helpful, so the test suites don't have to, in the off chance they don't use the workspace anyway.

File: test/AssemblySet_basic_test.py [hunk 6]
```diff
@@ -1,58 +1,22 @@
...
-        # do that outside this function..)
-        suffix = int(time.time() * 1000)
-        wsName = "test_SetAPI_" + str(suffix)
-        ret = cls.wsClient.create_workspace({'workspace': wsName})
-#        wsName = 'pranjan77:1477441032423'
-        cls.wsName = wsName
+        props = get_test_config()
+        for prop in ['cfg', 'ctx', 'serviceImpl', 'wsClient', 'wsName', 'wsURL']:
+            setattr(cls, prop, props[prop])
+
```
Comment (briehl) [inline test/AssemblySet_basic_test.py:19]: So much nicer! 🎉

File: test/test_config.py (hunk 5 above)
Comment (briehl) [reply test/test_config.py:54]: Oh, or maybe the tests could inherit from a baseclass that already does both the setup and teardown? Not necessary here, I don't think, though.

**Review by briehl — APPROVED** [review]
Just a couple questions and thoughts.

File: scripts/run_async.sh
Comment (ialarmedalien) [reply scripts/run_async.sh:1]: Yes, just changed it to +x , which recommitted it.

File: test/test_config.py (hunk 4 above)
Comment (ialarmedalien) [reply test/test_config.py:24]: Yep, will add in the next PR.

File: test/test_config.py (hunk 5 above)
Comment (ialarmedalien) [reply test/test_config.py:54]: I wanted to keep this PR fairly small. I'm debating whether to just shift the whole lot over to pytest and share the same fixtures for everything (the optimal approach, but will require a bit of search-and-replace).

**Pull request merged**